controllers/user_controller.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/drift-org/backend/models"
+	"github.com/gin-gonic/gin"
+	"github.com/kamva/mgm/v3"
+)
+
+type UserController interface {
+	Info(context *gin.Context)
+}
+
+type userController struct {
+}
+
+func NewUserController() UserController {
+	return &userController{}
+}
+
+func (ctrl *userController) Info(context *gin.Context) {
+	// Id refers to MongoDB user object ID. Specificty refers to the verbocity of the returned friends array
+	// 0 returns ONLY friend user IDs, 1 returns friend user IDs AND usernames
+	type ICreate struct {
+		Id          string `json:"id"`
+		Specificity int    `json:"specificity"`
+	}
+	var friendUsernames []string
+	var body ICreate
+
+	if err := context.ShouldBindJSON(&body); err != nil {
+		context.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	if body.Specificity > 1 || body.Specificity < 0 {
```

Comment:
Technically safer to do `body.Specificity != 1 && body.Specificity != 0`, since a specificity of 0.5 would pass your check.

Reply:
Unresolving. Actually, we might be able to get away with this entire thing.

According to https://chenyitian.gitbooks.io/gin-web-framework/content/docs/17.html, gin uses this validation library: https://pkg.go.dev/gopkg.in/go-playground/validator.v8#hdr-Baked_In_Validators_and_Tags.

So you should be able to validate 1 vs 0 in the binding. Potentially, something like "binding: eq=0|eq=1, required" (not sure), but you should experiment.

Reply:
Looks like validation can be included in binding. Keeping Specificity optional though so that requests with just ID default to specificity 0.